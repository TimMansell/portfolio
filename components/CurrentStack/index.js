import CurrentStack from './CurrentStack';

export default CurrentStack;
