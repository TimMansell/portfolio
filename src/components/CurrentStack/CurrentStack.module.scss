@import 'variables';
@import 'utilities/pxtorem';

.stack {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.item {
  display: flex;
  justify-content: center;
  align-self: center;
  padding: var(--vr-sm);
  width: 100%;

  &:last-of-type {
    margin-bottom: 0;
  }

  @media (min-width: $bp-sm) {
    margin-bottom: 0;
    width: auto;
  }
}
