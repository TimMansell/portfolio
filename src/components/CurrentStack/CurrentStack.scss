@import 'utilities/breakpoints';

.stack {
    display: grid;
    grid-gap: 32px;

    @include media-breakpoint-up(sm){
        grid-template-columns: repeat(4, 1fr);
    }

    @include media-breakpoint-up(md){
        grid-template-columns: repeat(6, 1fr);
    }

    @include media-breakpoint-up(xl){
        grid-template-columns: repeat(12, 1fr);   
    }

    &__item {
        align-self: center;    
        justify-self: center;   
    }
}