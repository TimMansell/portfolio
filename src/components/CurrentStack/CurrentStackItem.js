import React from 'react';
import PropTypes from 'prop-types';

import Image from '../Image';

import styles from './CurrentStack.module.scss';

export const CurrentStackItem = ({ url, name, imgWidth }) => {
  const img = name.toLowerCase();

  return (
    <div className={styles.item}>
      <a
        href={url}
        target="_blank"
        rel="noopener noreferrer"
        data-test="stack-item-link"
      >
        <Image
          src={`CurrentStack/img/${img}.svg`}
          alt={name}
          width={imgWidth}
        />
      </a>
    </div>
  );
};

CurrentStackItem.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  imgWidth: PropTypes.string.isRequired,
};

export default CurrentStackItem;
