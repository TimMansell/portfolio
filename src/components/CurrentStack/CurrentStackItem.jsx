import React from 'react';
import PropTypes from 'prop-types';

import Image from '../Image';

import styles from './CurrentStack.module.scss';

export const CurrentStackItem = ({ url, name, width, height }) => {
  const img = name.toLowerCase();

  return (
    <div className={styles.item}>
      <a
        href={url}
        target="_blank"
        rel="noopener noreferrer"
        data-test="stack-item-link"
      >
        <Image
          src={`CurrentStack/img/${img}.svg`}
          alt={name}
          width={width}
          height={height}
        />
      </a>
    </div>
  );
};

CurrentStackItem.propTypes = {
  url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
};

export default CurrentStackItem;
