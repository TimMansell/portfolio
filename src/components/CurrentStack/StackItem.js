import React from 'react';
import PropTypes from 'prop-types';

import SvgImage from '../SvgImage';

import styles from './CurrentStack.module.scss';

export const SkillsItem = ({ url, img, name, width }) => {
  const src = require(`./img/${img}`);

  return (
    <div className={styles.item}>
      <a href={url} target="_blank" rel="noopener noreferrer">
        <SvgImage src={src} name={name} width={width} />
      </a>
    </div>
  );
};

SkillsItem.propTypes = {
  url: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
};

export default SkillsItem;
