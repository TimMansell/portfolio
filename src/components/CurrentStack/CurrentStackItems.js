import React from 'react';

import CurrentStackItem from './CurrentStackItem';

import styles from './CurrentStack.module.scss';

import stackItems from './json/stack.json';

export const CurrentStackItems = () => {
  return (
    <div className={styles.stack} data-test="stack-items">
      {stackItems.map((stack, i) => (
        <CurrentStackItem {...stack} key={i} />
      ))}
    </div>
  );
};

export default CurrentStackItems;
