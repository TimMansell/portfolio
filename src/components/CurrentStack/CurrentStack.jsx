import Header from 'components/Header';
import CurrentStackItems from './CurrentStackItems';

export const CurrentStack = () => {
  return (
    <>
      <Header
        title="Current Stack"
        text="I spend most of my time with these technologies / tools / methodologies"
        type="primary"
      />

      <CurrentStackItems />
    </>
  );
};

export default CurrentStack;
