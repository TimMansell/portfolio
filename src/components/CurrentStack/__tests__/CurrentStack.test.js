import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CurrentStack from '../CurrentStack';

// jest.mock(
//   '../json/stack.json',
//   () => [
//     {
//       name: 'Item 1',
//       url: 'url',
//       img: 'react.svg',
//       width: '65',
//     },
//     {
//       name: 'Item 2',
//       url: 'url',
//       img: 'nodejs.svg',
//       width: '90',
//     },
//   ],
//   { virtual: true }
// );

describe('CurrentStack', () => {
  it('should render my component', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<CurrentStack />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<CurrentStack />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  // it('should render correct amount of items', () => {
  //   const wrapper = shallow(<CurrentStack />);
  //   const list = wrapper.find('[data-test="stack-items"]');

  //   expect(list.children()).toHaveLength(2);
  // });
});
