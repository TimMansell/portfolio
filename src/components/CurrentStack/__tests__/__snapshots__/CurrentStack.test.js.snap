// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentStack should match snapshot 1`] = `
Array [
  <div
    className="heading headingPrimary"
  >
    <h2
      aria-label="Current Stack"
      className="title"
      data-test="heading-title"
    >
      Current Stack
    </h2>
    <p
      data-test="heading-description"
    >
      I spend most of my time with these technologies / tools / methodologies
    </p>
  </div>,
  <div
    className="stack"
    data-test="stack-items"
  >
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://reactjs.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="React"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "react.svg",
            }
          }
          title="React"
          width="65"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://nodejs.org/en/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="NodeJS"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "nodejs.svg",
            }
          }
          title="NodeJS"
          width="90"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://vuejs.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="VueJS"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "vuejs.svg",
            }
          }
          title="VueJS"
          width="60"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://www.npmjs.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="NPM"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "npm.svg",
            }
          }
          title="NPM"
          width="90"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://webpack.js.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Webpack"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "webpack.svg",
            }
          }
          title="Webpack"
          width="75"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://yarnpkg.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Yarn"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "yarn.svg",
            }
          }
          title="Yarn"
          width="130"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://eslint.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="ESLint"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "eslint.svg",
            }
          }
          title="ESLint"
          width="75"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://jestjs.io/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Jest"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "jest.svg",
            }
          }
          title="Jest"
          width="50"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://babeljs.io/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Babel"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "babel.svg",
            }
          }
          title="Babel"
          width="140"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://www.cypress.io/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Cypress"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "cypress.svg",
            }
          }
          title="Cypress"
          width="60"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://sass-lang.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="SCSS"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "sass.svg",
            }
          }
          title="SCSS"
          width="70"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://code.visualstudio.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="VSCode"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "vscode.svg",
            }
          }
          title="VSCode"
          width="55"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://git-scm.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Git"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "git.svg",
            }
          }
          title="Git"
          width="105"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://postcss.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="PostCSS"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "postcss.svg",
            }
          }
          title="PostCSS"
          width="60"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://www.netlify.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Netlify"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "netlify.svg",
            }
          }
          title="Netlify"
          width="140"
        />
      </a>
    </div>
    <div
      className="item"
    >
      <a
        data-test="stack-item-link"
        href="https://graphql.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="GraphQL"
          className="img"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "graphql.svg",
            }
          }
          title="GraphQL"
          width="55"
        />
      </a>
    </div>
  </div>,
]
`;
